### PR TITLE
Updating CNTKLearner to work with latest gpu branch updated with latest master

### PR DESCRIPTION
**Pull request by imatiach-msft**
Updating CNTKLearner to work with latest gpu branch updated with latest master.
Removed setting owner, fixed ByteArrayParam transfer from java to python, fixed path to model.
Verified full end-to-end scenario.

### CODE REVIEW

File: src/cntk-train/src/main/scala/CommandBuilders.scala [hunk 1]
```diff
@@ -15,6 +15,7 @@ import org.apache.spark.sql.SparkSession
 import org.slf4j.Logger
 
 import scala.sys.process._
```
Comment (tongwen11) [inline src/cntk-train/src/main/scala/CommandBuilders.scala]: This import line is duplicated, there is one above.

Comment (imatiach-msft) [inline src/cntk-train/src/main/scala/CommandBuilders.scala]: removed duplicate

**Comment by imatiach-msft** [comment]
updated to remove duplicate

File: src/cntk-train/src/main/python/CNTKLearner.py [hunk 2]
```diff
@@ -8,11 +8,31 @@
...
 
 @inherit_doc
 class CNTKLearner(_CNTKLearner):
     """
+    Transforms the params from java object, with special case for byte array
+
+    Args:
+        java_model (CNTKModel): The python CNTKModel to transfer parameters to from java
+    """
+    def _custom_transfer_params_from_java(self, model):
```
Comment (mhamilton723) [inline src/cntk-train/src/main/python/CNTKLearner.py]: Is there a way to edit the main transfer params from java so we can make it more robust to these things. There will be many more custom parameters in the future

File: src/cntk-train/src/main/scala/CNTKLearner.scala [hunk 3]
```diff
@@ -154,9 +154,10 @@ class CNTKLearner(override val uid: String) extends Estimator[CNTKModel] with CN
     val output = cb.runCommand
     logInfo("CNTK training finished")
     conformedData.unpersist()
+    val modelPath = "file://" + config.getModelPath
```
Comment (mhamilton723) [inline src/cntk-train/src/main/scala/CNTKLearner.scala]: SHould this be automatic? What about when the model is on Wasb?

File: src/cntk-train/src/main/scala/CommandBuilders.scala [hunk 4]
```diff
@@ -133,9 +132,7 @@ class MPICommandBuilder(log: Logger,
     val outputPath = new Path(s"file:///${identityDir.getAbsolutePath}/identity")
     // Copy from wasb to local file
     using(Seq(inputPath.getFileSystem(hadoopConf))) { fs =>
```
Comment (mhamilton723) [inline src/cntk-train/src/main/scala/CommandBuilders.scala]: I added a using with one object so no need for the Seq wrapper

File: src/core/serialize/src/main/scala/params/ByteArrayParam.scala [hunk 5]
```diff
@@ -12,4 +14,15 @@ class ByteArrayParam(parent: Params, name: String, doc: String, isValid: Array[B
     def this(parent: Params, name: String, doc: String) =
       this(parent, name, doc, ParamValidators.alwaysTrue)
 
-  }
+    /** Creates a param pair with a java.util.List of values (for Java and Python). */
+    def w(value: java.util.List[java.lang.Byte]): ParamPair[Array[Byte]] =
+      w(value.asScala.map(_.asInstanceOf[Byte]).toArray)
+
```
Comment (mhamilton723) [inline src/core/serialize/src/main/scala/params/ByteArrayParam.scala]: I dont want to json encode these as I think it causes memory problems with large models

**Comment by tongwen11** [comment]
Validated the cifar10 notebook run in a freshly deployed environment with success.

File: src/cntk-train/src/main/scala/CNTKLearner.scala (hunk 3 above)
Comment (imatiach-msft) [inline src/cntk-train/src/main/scala/CNTKLearner.scala]: for CNTKLearner it is currently always local, but you're right eventually we want to support both once we have the parquet reader working.

File: src/cntk-train/src/main/python/CNTKLearner.py (hunk 2 above)
Comment (imatiach-msft) [inline src/cntk-train/src/main/python/CNTKLearner.py]: discussed, the code is a bit more convoluted than we thought, we will keep this as is for now

File: src/cntk-train/src/main/scala/CommandBuilders.scala (hunk 4 above)
Comment (imatiach-msft) [inline src/cntk-train/src/main/scala/CommandBuilders.scala]: nice, I've removed the Seq()

File: src/core/serialize/src/main/scala/params/ByteArrayParam.scala (hunk 5 above)
Comment (imatiach-msft) [inline src/core/serialize/src/main/scala/params/ByteArrayParam.scala]: removed the jsonEncode/Decode as they were not being used anyway to fix this specific issue

**Comment by tongwen11** [comment]
Also validated that in the client mode, the set owner operation is not needed.

File: src/cntk-train/src/main/python/CNTKLearner.py [hunk 6]
```diff
@@ -8,11 +8,31 @@
...
+        for param in model.params:
+            if model._java_obj.hasParam(param.name):
+                java_param = model._java_obj.getParam(param.name)
+                if model._java_obj.isSet(java_param):
+                    param_val = model._java_obj.getOrDefault(java_param)
+                    if isinstance(param_val, (bytearray, bytes)):
```
Comment (drdarshan) [inline src/cntk-train/src/main/python/CNTKLearner.py]: See my other comment on Mark's PR #98 - checking ``isinstance`` is generally not preferred in Python. 
Also, does this code work in both 2.7 and 3.6?

File: src/cntk-train/src/main/scala/CNTKLearner.scala (hunk 3 above)
Comment (drdarshan) [inline src/cntk-train/src/main/scala/CNTKLearner.scala]: +1. Can't we make config.getModelPath return the correct URI (local vs. HDFS vs. WASB etc)

**Pull request merged**